preliminary work for SoundTouch BPM detection

enetheru opened the pull request:
removed dependency on aubio

This pull request is just so you know i'm looking into it. I will update it as i get further along.

enetheru committed:
preliminary work for SoundTouch BPM detection

 removed dependency on aubio

basisbit commented:
you might want to check the licensing stuff. I am not sure if one is allowed to use a LGPL library in a GPLv2+ application. If that actually is allowed, a link to some good explaining article would be nice 😃

enetheru committed:
created BPMDetect derived from QAudioDecoder

a very light shim over the top of QAudioDecoder interfacing it with SoundTouch::BPMDetect class

enetheru commented:
@basisbit lgpl can be used in proprietary software so long as you arent statically linking to it. if it remains a separate library you're sweet and is the entire purpose of LGPL vs GPL

https://stackoverflow.com/questions/10130143/gpl-lgpl-and-static-linking is no1 on google.

enetheru commented:
also the QT libraries are lgplv3 so thats already happening.. :P

enetheru committed:
updated travis.yml for CI testing

enetheru committed:
created bpmDetected signal and connected it to the ui

enetheru committed:
missed semicolon in travis.yml

enetheru committed:
trying to fix travis.yml

includes of soundtouch dont work on OSX
qtmultimedia5-dev needed

basisbit commented:
the linked stackoverflow thread does not answer my question at all. UltraStar Creator is licensed under GPLv2+, SoundTech is licensed under LGPL. Please try again.

enetheru commented:
@basisbit  I'm sorry that you dont think that answer's your question. Since you are the one who is unsure please go and do the research yourself. I however, am sure that it's perfectly legal to link to an LGPL library from anywhere so long as it's only linked dynamically it will not pollute the rest of the source. the terms of ultrastars licence has zero bearing on SoundTouch's code since they do not have any dependency or link to us, that's not how licensing requirements work.

AlexanderS commented:
https://softwareengineering.stackexchange.com/a/342158
